src/devices/secplus_v1.c

```diff
@@ -0,0 +1,432 @@
+    uint32_t rolling_temp = 0; // max 2**32
+    uint32_t fixed        = 0; // max 3^20 ( ~32 bits )
+
+    uint8_t *res;
+    res = result_1;
+    res++;
+
+    uint32_t acc = 0;
+    for (int i = 0; i < 20; i += 2) {
+        uint8_t digit = 0;
+
+        digit        = res[i];
+        rolling_temp = (rolling_temp * 3) + digit;
+        acc += digit;
+
+        digit = (60 + res[i + 1] - acc) % 3;
+        fixed = (fixed * 3) + digit;
+        acc += digit;
+    }
+
+    res = result_2;
+    res++;
+
+    acc = 0;
+    for (int i = 0; i < 20; i += 2) {
+        uint8_t digit = 0;
+
+        digit        = res[i];
+        rolling_temp = (rolling_temp * 3) + digit;
+        acc += digit;
+
+        digit = (60 + res[i + 1] - acc) % 3;
+        fixed = (fixed * 3) + digit;
+        acc += digit;
+    }
+
+    rolling = reverse32(rolling_temp);
+
+    /*
+        we now have values for rolling & fixed
+        next we extract status info stored in the value for 'fixed'
+    */
+    int switch_id  = fixed % 3;
+    int id0        = (fixed / 3) % 3;
+    int id1        = (int)(fixed / 9) % 3;
+    int pad_id     = 0;
+    int pin        = 0;
+    char pin_s[24] = {0};
+
+    int remote_id = 0;
+    char *button  = "";
+
+    if (id1 == 0) {
+        //  pad_id = (fixed // 3**3) % (3**7)     27  3^72187
+        pad_id = (fixed / 27) % 2187;
+        // pin = (fixed // 3**10) % (3**9)  3^10= 59049 3^9=19683
+        pin = (fixed / 59049) % 19683;
+
+        if (0 <= pin && pin <= 9999) {
+            snprintf(pin_s, sizeof(pin_s), "%04d", pin);
+        }
+        else if (10000 <= pin && pin <= 11029) {
+            strcat(pin_s, "enter");
+        }
+
+        int pin_suffix = 0;
+        // pin_suffix = (fixed // 3**19) % 3   3^19=1162261467
+        pin_suffix = (fixed / 1162261467) % 3;
+
+        if (pin_suffix == 1)
+            strcat(pin_s, "#");
+        else if (pin_suffix == 2)
+            strcat(pin_s, "*");
+
+        // if (decoder->verbose)
+        //     fprintf(stderr, "pad_id=%d\tpin=%d\tpin_s=%s\n", pad_id, pin, pin_s);
+    }
+    else {
+        remote_id = (int)fixed / 27;
+        if (switch_id == 1)
+            button = "left";
+        else if (switch_id == 0)
+            button = "middle";
+        else if (switch_id == 2)
+            button = "right";
+
+        // if (decoder->verbose)
+        //     fprintf(stderr, "remote_id=%d\tbutton=%s\n", remote_id, button);
+    }
+
+    // preformat unsigned int
+    char rolling_str[16];
+    snprintf(rolling_str, sizeof(rolling_str), "%u", rolling);
+
+    // preformat unsigned int
+    char fixed_str[16]; // should be 10 chars max
+    snprintf(fixed_str, sizeof(fixed_str), "%u", fixed);
+
+    // fprintf(stderr,  "# Security+:  rolling=2320615320  fixed=1846948897  (id1=2 id0=0 switch=1 remote_id=68405514 button=left)\n");
+    /* clang-format off */
+    data_t *data;
+    data = data_make(
+            "Model",       "Model",    DATA_STRING, "Secplus_v1",
```

Comment:
should be "model" (lowercase). Is there some kind of (composite) "id" we can use?

Reply:
"Model" -> "model" : FIxed.

I have no idea about a composite ID (@argilo ?)

Reply:
I'm not sure what a "composite id" means in this context.

Reply:
The protocol itself is named Security+ and it it used in many devices sold under various brand names.

Reply:
Official names go in the r_device name field. We have a commitment to only output sanitized model names (of roughly `[A-Za-z][A-Za-z0-9]*(-[A-Za-z0-9]+)?`) -- esp. `+` is really bad for MQTT.

Reply:
I mean just any cobbled together id, as `id0`, `id1` are not good for that. Is `remote_id` a fixed unique id?

Reply:
I guess a ID _maybe_ can be derived from the  `fixed` data part  if all the flags bits were masked or truncated 

 @argilo what do you think ?

Reply:
`remote_id` (for pushbutton remotes) and `pad_id` (for keypad remotes) are different for every remote. They don't seem to tell you anything about the model.

Reply:
Looking at the other model names, it seems that spaces are allowed. If we wanted to be more descriptive, we could use "Security Plus remote", or even distinguish between the two types "Security Plus pushbutton remote" and "Security Plus PIN pad".

Reply:
I think what @zuckschwerdt  was asking for was a unique identifier for the `Id` field so  different remotes can be easily  distinguishable .

Reply:
Only alphanum and a single dash in the middle -- it's a short unique key, useable as database key, nothing descriptive -- e.g. we use the model as MQTT topic. If the same protocol has different sensors with different functions (not sub-function of the same sensor) that should be given as `subtype`, here maybe "pad" and "remote". S.a. https://triq.org/rtl_433/DATA_FORMAT.html#message-data
If remote_id and pad_id are mutually exclusive the output should just be `id` with the appropriate value.

Reply:
ok I added `id` field that is equal to `remote_id` or `pad_id` depending on the data.

Reply:
> If remote_id and pad_id are mutually exclusive the output should just be id with the appropriate value.

They are, in the sense that you've either got a `remote_id` (for pushbutton remotes) or `pad_id` (for PIN pads). Agreed that they may as well both go in a single "id" field.